lib/game_setting/game_setting.rb

```diff
@@ -0,0 +1,28 @@
+require_relative '../validator'
+require_relative '../messages'
+require_relative './setting_types'
+
+class GameSetting
+    attr_accessor :setting, :input
+
+    def initialize(setting:)
+        @setting = setting
+    end
+
+    def valid?(input: nil)
+        @input = input
```

Comment:
If you never return `@input` and you never use it within the class, why have it? Can't this just be `setting.valid?(input: input)`?

Reply:
you're right! I think I was trying something and forgot to remove this